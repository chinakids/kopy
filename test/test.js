import test from 'ava'
import copy from '../src'
import {spawnSync} from 'child_process'
import rm from 'rimraf'

test.after('cleanup', () => {
  rm.sync('./dest*')
})

test('main', async t => {
  const files = await copy('./fixture-src', './dest', {
    data: {
      has: true,
      name: 'hi'
    }
  })
  const res = require('./dest/hi')
  t.is(res.name, 'hi')
  t.true(res.has)
  t.false(res.hasNot)

  // check files
  t.deepEqual(files, ['hi.json', 'skipInterpolation.json', 'deep/bye.json'])

  const foo = require('./dest/deep/bye')
  t.is(foo.name, 'hi')
})

test('it should skip interpolation by glob patterns', async t => {
  const files = await copy('./fixture-src', './dest-skip', {
    data: {
      name: 'hi'
    },
    skipInterpolation: [
      'skipInterpolation.json',
      'deep/bye.*'
    ]
  })
  const foo = require('./dest-skip/skipInterpolation')
  const bar = require('./dest-skip/deep/bye')
  t.is(foo.name, '{{ name }}')
  t.is(bar.name, '{{ name }}')
})
