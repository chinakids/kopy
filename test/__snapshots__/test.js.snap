// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generated files 1`] = `
Array [
  "ab.json",
  "abr.json",
  "foo.json",
]
`;
